src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback } from "react";
+import { useKeyEvent } from "hooks";
+import Clickable from "../Clickable/Clickable";
+import Flex from "../Flex/Flex";
+import Avatar from "../Avatar/Avatar";
+import avatarGroupCounterTooltipContentStyles from "./AvatarGroupCounterTooltipContent.module.scss";
+
+const KEYS = ["Tab"];
+
+export function useTooltipContentTabNavigation({
+  counterContainerRef = undefined,
+  tooltipContentContainerRef,
+  focusPrevPlaceholderRef,
+  focusNextPlaceholderRef,
+  setShouldUpdate,
+  setIsTooltipVisible
+}) {
+  const hideTooltip = useCallback(() => {
+    // Tricky way to close the tooltip
+    setTimeout(() => {
+      setIsTooltipVisible(false);
+      setIsTooltipVisible(true);
+    });
+  }, [setIsTooltipVisible]);
+
+  // For Counter
+  useKeyEvent({
+    keys: KEYS,
+    withoutAnyModifier: true,
+    ref: counterContainerRef,
+    callback: useCallback(
+      e => {
+        if (e.target === counterContainerRef.current) {
+          e.preventDefault();
+          tooltipContentContainerRef.current && tooltipContentContainerRef.current.focus();
+          setShouldUpdate(prev => !prev);
+        }
+      },
+      [counterContainerRef.current, setShouldUpdate, tooltipContentContainerRef]
+    )
+  });
+
+  // For Counter
+  useKeyEvent({
+    keys: KEYS,
+    modifier: useKeyEvent.modifiers.SHIFT,
+    ref: counterContainerRef,
+    callback: useCallback(
```

Comment:
all of the `useKeyEvents` are almost the same, can you please extract to a "customHook" ?

Reply:
Don't see how you can effectively avoid `useKeyEvents` usage duplication... Are you speaking about this? 
Or you mean just extracting `useTooltipContentTabNavigation` into a separate file? Cause they are already in a custom hook